Added support for converting units without quantity search tag.

ZhichaoHong opened the pull request:
Need to review with Adir.

Commit:
Added support for converting units without quantity search tag.

adirh3 commented on UnitsConverter.Fluent.Plugin/ConvertUtil.cs:
```diff
@@ -27,6 +27,82 @@ internal class ConvertUtil
             QuantityType.Volume,
         };
 
+        public static List<IQuantity> ConvertWithoutQuantity(string text)
```
I suggest making this `bool TryConvertWithQuantity(string text, out List<IQuantity> quantity)` and then inline it with the `if` in the search app, line 113.

ZhichaoHong replied:
I see, you prefer returning the bool instead of using empty list to represent conversion failure?

adirh3 commented on UnitsConverter.Fluent.Plugin/ConvertUtil.cs:
```diff
@@ -27,6 +27,82 @@ internal class ConvertUtil
+            string from = parts[0].Trim();
+            string to = parts.Length > 1 ? parts[1].Trim() : null;
+            
+            if (Acceleration.TryParse(from, out Acceleration accel))
```
If `QuantityType` names are matching all the other enums you can do something like this -
`new List<Enum> { typeof(Area), typeof(Duration), ...}.FirstOrDefault(t => Enum.TryParse(t, a, out var _) && QuantityType.TryParse(t.Name, out quantityType)?.Name`
If you do keep that list as a static field.

ZhichaoHong replied:
Adir, I have removed this class and add all logic in UnitHandler. I may need to add some of the logic back to support this term "5 meters in" without a given "toUnit".

adirh3 commented on UnitsConverter.Fluent.Plugin/UnitsConversionSearchApp.cs:
```diff
@@ -109,6 +109,17 @@ public async IAsyncEnumerable<ISearchResult> SearchAsync(SearchRequest searchReq
 
             QuantityType quantityType = QuantityType.Undefined;
+
+            if (searchedTag.Equals(UnitsConverterSearchTag) && !string.IsNullOrWhiteSpace(searchedText))
```
Just to be sure - when I said without a search tag, I meant without using any search tags.
So users can search something like `5m in seconds`, but I am not sure if full terms are supported, like `seconds` instead of `s`.

ZhichaoHong replied:
Does this mean the UnitConversionSearchTag is optional? So the conversion can work with only the searchedText? Please clarify. I have not thought about this.

Commit:
Refactored the logic to work without Quantities search tags.

Commit:
Added support to search all using "..." as optional input in the searchedText.

Commit:
Added score to prioritize the given unit in the searchedText. Added MLFeatures in the UnitsConversionSearchResult.

Commit:
Added pluginsInfo.json for packaging needs.

Commit:
Added support for searching with search tag if full search term (i.e. both from and to unit are provided).

If search tag (unitsconverter) is used, from unit is only needed. In this situation, all converted units are displayed.

Commit:
Filter out the from unit when partial match is executed.

Commit:
Update version to 1.0.2.0

Commit:
Added more copy search operations so different value & unit combinations can be selected for converted result.

Make more examples and show the 2 random examples.
